compiler/rustc_parse/src/parser/attr.rs

```diff
@@ -250,14 +250,57 @@ impl<'a> Parser<'a> {
     ///     PATH `{` TOKEN_STREAM `}`
     ///     PATH
     ///     PATH `=` UNSUFFIXED_LIT
+    ///     `unsafe` `(` PATH `(` TOKEN_STREAM `)` `)`
+    ///     `unsafe` `(` PATH `[` TOKEN_STREAM `]` `)`
+    ///     `unsafe` `(` PATH `{` TOKEN_STREAM `}` `)`
+    ///     `unsafe` `(` PATH                      `)`
+    ///     `unsafe` `(` PATH `=` UNSUFFIXED_LIT   `)`
     /// The delimiters or `=` are still put into the resulting token stream.
     pub fn parse_attr_item(&mut self, capture_tokens: bool) -> PResult<'a, ast::AttrItem> {
         maybe_whole!(self, NtMeta, |attr| attr.into_inner());
 
+        let do_parse = |this: &mut Self| {
+            let is_unsafe = this.eat_keyword(kw::Unsafe);
+            let unsafety = if is_unsafe {
+                let unsafe_span = this.prev_token.span;
+                this.psess.gated_spans.gate(sym::unsafe_attributes, unsafe_span);
+                this.expect(&token::OpenDelim(Delimiter::Parenthesis))?;
+
+                ast::Unsafe::Yes(unsafe_span)
+            } else {
+                ast::Unsafe::No
+            };
+
+            let path = this.parse_path(PathStyle::Mod)?;
```

Comment:
I expect this to result in a slightly weird error message when doing something like `#[unsafe(unsafe(no_mangle))]`. Can we add a test case for that?

Reply:
```
error: expected identifier, found keyword `unsafe`
  --> /home/quydx/code/rust/tests/ui/attributes/unsafe/double-unsafe-in-attribute.rs:1:10
   |
LL | #[unsafe(unsafe(no_mangle))]
   |          ^^^^^^ expected identifier, found keyword
   |
help: escape `unsafe` to use it as an identifier
   |
LL | #[unsafe(r#unsafe(no_mangle))]
   |          ++

error: cannot find attribute `r#unsafe` in this scope
  --> /home/quydx/code/rust/tests/ui/attributes/unsafe/double-unsafe-in-attribute.rs:1:10
   |
LL | #[unsafe(unsafe(no_mangle))]
   |          ^^^^^^
```
The error is basically just what we have now. I can probably add a note saying that `unsafe(...)` doesn't nest.